doc/contribute/templates_styleguides/pynest_example_template.py

```diff
@@ -0,0 +1,150 @@
+----------------------------------------------------------------
+[[ Titles should be one line and state what the example does.
+  It should begin with a verb in the present tense and include type of model
+  and/or method]]
+
+[[ Extended summary - a detailed explanation of your example. Try to answer the
+   folowing questions. ]]
+[[ What does this script do? What is the purpose? ]]
+This template demonstrates how to create an example Python script for
+NEST.
+Copy this file and replace the sample text with a description of your
+example script.
+
+Make sure to remove any text that is irrelevant for your example!
+
+The format is based on `NumPy style docstring
+<https://numpydoc.readthedocs.io/en/latest/format.html>`_ and uses
+reStructuredText markup. Please review the syntax rules if you are
+unfamiliar with either reStructuredText or NumPy style docstrings.
+Your example should contain a complete code-block that begins with all
+necessary imports and ends with code that displays the output.
+
+Below is a more concrete example of how an extended summary could look like:
+
+This script simulates a neuron by an excitatory and an inhibitory
+population of neurons firing a Poisson spike train.
+Optimization is performed using the `bisection` method from Scipy,
+which simulates the network repeatedly.
+
+The aim of this example script is to find a firing rate for the inhibitory
+population that will make the neuron fire at the same rate as the excitatory
+population.
+
+[[ What kind of output is expected? ]]
+
+The output shows the target neuron's membrane potential as a function of time.
+
+[[ Does this example have a real world application or use case?
+  Are there particular applications or areas of research that would benefit
+  from this example? You can reference relevant papers that this example
+  may be based on]]
+
+This model used here is applicable for neurorobotics, particularly cases of ...
+
+[[ If applicable, state any prerequisite the reader needs to have installed or
+   configured that is not standard ]]
+
+Please ensure that you have configured MUSIC to be ON in your NEST
+configuration:
+``cmake -Dwith-music=[ON</path/to/music>]``
+
+[[ If applicable, mention the literature reference for this example.
+   Note the syntax of the citation. And don't forget to add a "References"
+   section! ]]
+
+This model used here corresponds to the formulation presented in
+Sander et al. [1]_ and the bisection method developed in
+Gewaltig and Diesmann [2]_.
+
+[[ See Also section - Include a couple of related examples, models,
+   or functions. ]]
+
+See Also
+---------
+:ref:`Intrinsic current subthreshold <label_name>`
+:doc:`some other doc </path/to/filename>`
+
+Notes
+------
+[[ Additional information can be included here regarding background theory,
+   relevant mathematical equations etc. ]]
+
+The value of :math:`\omega` is X.
+For the population and time-averaged from the spiking simulation:
```

Comment:
This sentence doesn't quite make sense to me. It might be that it connects with the sentence above, so that it could be
```
The value of :math:`\omega` is X
for the population. Time-averaged from the spiking simulation gives:
```
Or, I think it would be ok to drop the first part of the sentence, so that it becomes
```
The value of :math:`\omega` is X.
Time-averaged from the spiking simulation:
```

Reply:
```suggestion
Time-averaged from the spiking simulation:
```